### PR TITLE
[aievec] Fix ups to llvmir conversion

### PR DESCRIPTION
**Pull request by jsetoain**
The current version fails to convert with multi-dimensional vectors.

### CODE REVIEW

**Comment by jsetoain** [comment]
@makslevental you may want to cherry-pick this one

File: test/Conversion/AIEVecToLLVM/test-ups.mlir [hunk 1]
```diff
@@ -182,3 +182,17 @@ func.func @v32f32_ups_v32bf16(%arg0 : vector<32xbf16>) {
...
+
+// -----
+
+// CHECK-LABEL: @multidim_ups_i8_to_i32
+// CHECK-SAME: %[[V:.*]]: vector<4x8xi8>
+// CHECK: %[[FV:.*]] = vector.shape_cast %[[V]] : vector<4x8xi8> to vector<32xi8>
+// CHECK: %[[FUPS:.*]] = "xllvm.intr.aie2.acc32.v32.I256.ups"(%[[FV]],
+// CHECK-SAME: %{{.*}}, %{{.*}}) : (vector<32xi8>, i32, i32) -> vector<16xi64>
+// CHECK: %[[FR:.*]] = llvm.bitcast %3 : vector<16xi64> to vector<32xi32>
+// CHECK: %[[UPS:.*]] = vector.shape_cast %[[FR]] : vector<32xi32>
```
Comment (jamestcl-amd) [inline test/Conversion/AIEVecToLLVM/test-ups.mlir:194]: missing `to vector<4x8xi32>` ?

**Review by jamestcl-amd — APPROVED** [review]
LGTM

**Comment by newling** [comment]
> @makslevental you may want to cherry-pick this one

I have a branch accumulating the changes we need, I'll copy this change onto the branch (which I'll eventually land). Unfortunately we can't just cp the file file anymore because they have they've diverged.

**Pull request merged**